projects/plugins/jetpack/extensions/blocks/blogging-prompt/editor.js

```diff
@@ -13,36 +13,53 @@ import save from './save';
 import './editor.scss';
 import './style.scss';
 
-registerJetpackBlockFromMetadata( metadata, {
-	edit,
-	save,
-	example: {
-		attributes: {
-			answersLink: 'https://wordpress.com/tag/dailyprompt',
-			answersLinkText: __( 'View all responses', 'jetpack' ),
-			gravatars: [ { url: avatar1 }, { url: avatar2 }, { url: avatar3 } ],
-			promptLabel: __( 'Daily writing prompt', 'jetpack' ),
-			promptText: __( "What's your favorite place to visit?", 'jetpack' ),
-			promptFetched: true,
-			promptId: 1234,
-			showResponses: true,
-			showLabel: true,
-			tagsAdded: true,
-			isBloganuary: false,
-		},
-	},
-} );
+// Register the block and return a promise that resolves when registration is complete
+const registerBlock = () =>
+	new Promise( resolve => {
+		registerJetpackBlockFromMetadata( metadata, {
+			edit,
+			save,
+			example: {
+				attributes: {
+					answersLink: 'https://wordpress.com/tag/dailyprompt',
+					answersLinkText: __( 'View all responses', 'jetpack' ),
+					gravatars: [ { url: avatar1 }, { url: avatar2 }, { url: avatar3 } ],
+					promptLabel: __( 'Daily writing prompt', 'jetpack' ),
+					promptText: __( "What's your favorite place to visit?", 'jetpack' ),
+					promptFetched: true,
+					promptId: 1234,
+					showResponses: true,
+					showLabel: true,
+					tagsAdded: true,
+					isBloganuary: false,
+				},
+			},
+		} );
+
+		// Wait for next tick to ensure registration is complete
+		setTimeout( resolve, 0 );
+	} );
 
 async function insertTemplate( promptId ) {
 	await waitForEditor();
 
+	// Ensure block is registered before insertion
+	const blockType = getBlockType( 'jetpack/blogging-prompt' );
+	if ( ! blockType ) {
+		await registerBlock();
+	}
+
 	const { insertBlocks } = dispatch( 'core/block-editor' );
-	const bloggingPromptBlocks = [
-		createBlock( 'jetpack/blogging-prompt', { promptFetched: false, promptId, tagsAdded: true } ),
-		createBlock( 'core/paragraph' ),
-	];
 
-	insertBlocks( bloggingPromptBlocks, 0, undefined, false );
+	const bloggingPromptBlock = createBlock( 'jetpack/blogging-prompt', {
```

Comment:
This just some cleanup so that we have two distinct `createBlock` calls. I can remove if anyone feels strongly about it.